pkg/target/target_test.go

```diff
@@ -693,13 +695,74 @@ func TestMatcher_Match(t *testing.T) {
 	}
 }
 
+func TestNamespaceCache(t *testing.T) {
+	ns1 := makeNamespace("my-ns1", map[string]string{"ns1": "label"})
```

Comment:
Make Namespaces inline in the tests (`"myns2": makeNamespace(...)`). This is both for clarity (you can understand what a test case is doing just by looking at the test case) and to ensure tests can't have cross-talk as they're guaranteed to be modifying their own instances of the Namespace.